### PR TITLE
User page comments

### PR DESCRIPTION
**Pull request by Munch-Z**

### CODE REVIEW

File: models/Comment.ts [hunk 1]
```diff
@@ -22,17 +22,17 @@ export interface ICommentModel extends Model<ICommentDocument> {
 export class CommentData {
   _id: string;
   dateCreated: Date;
-  author: UserData;
+  author: UserData | 'No author';
```
Comment (brandtkstrom) [inline models/Comment.ts]: I believe you need double pipes for logical OR here

File: models/User.ts [hunk 2]
```diff
@@ -8,21 +9,24 @@ export class UserData {
   follows: UserData[];
   followers: UserData[];
   favorites: IProduct[];
-  comments: IComment[];
+  comments: ICommentDocument[] | CommentData[];
```
Comment (brandtkstrom) [inline models/User.ts]: We should use 1 type here

File: models/Comment.ts (hunk 1 above)
Comment (Munch-Z) [reply models/Comment.ts]: TypeScript screamed at me when I tried to make this change, so I'm gonna assume it's correct.

File: models/User.ts (hunk 2 above)
Comment (Munch-Z) [reply models/User.ts]: Fixed

**Pull request merged**